doc/source/tune/api_docs/suggestion.rst

```diff
@@ -72,6 +72,54 @@ Tune also provides helpful utilities to use with Search Algorithms:
  * :ref:`repeater`: Support for running each *sampled hyperparameter* with multiple random seeds.
  * :ref:`limiter`: Limits the amount of concurrent trials when running optimization.
 
+Saving and Restoring
+--------------------
+
+Certain search algorithms have ``save/restore`` implemented,
+allowing reuse of learnings across multiple tuning runs.
+
+.. code-block:: python
+
+    search_alg = HyperOptSearch()
+
+    experiment_1 = tune.run(
+        trainable,
+        search_alg=search_alg)
+
+    search_alg.save("./my-checkpoint.pkl")
+
+    # Restore the saved state onto another search algorithm
+
+    search_alg2 = HyperOptSearch()
+    search_alg2.restore("./my-checkpoint.pkl")
+
+    experiment_2 = tune.run(
+        trainable,
+        search_alg=search_alg2)
+
+Further, Tune automatically saves its state inside the current experiment folder ("Result Dir") during tuning.
+
+Note that if you have two Tune runs with the same experiment folder,
+the previous state checkpoint will be overwritten. You can
+avoid this by making sure ``tune.run(name=...)`` is set to a unique
+identifier.
+
+.. code-block:: python
+
+    search_alg = HyperOptSearch()
+    experiment_1 = tune.run(
+        cost,
+        num_samples=5,
+        search_alg=search_alg,
+        verbose=0,
+        name="my-experiment-1",
+        local_dir="~/my_results")
+
+    search_alg2 = HyperOptSearch()
+    Searcher.restore_from_dir(
```

Comment:
Why have two ways to do this? Either the search_alg can restore directly or through the Searcher.

Reply:
It just complicates things IMO.

Reply:
I'm not sure what you're suggesting. Can you provide some code examples?

Reply:
why not just `search_alg2.restore_from_dir(os.path.join("~/my_results/my-experiment-1"))`?

Reply:
> It just complicates things IMO.

Also, I'm happy to clarify the design decision, but I'm not sure what problems you're seeing nor the actionable item to take here. 

It'd be great if during review, the comments would clarify 1. the problem with the current solution and 2. if possible, propose something that addresses the state problems while meeting existing requirements.

Reply:
I just wanted to understand the reasoning behind why there is 2 ways to do (what seems like) the same thing. To restore from a path we can do it directly from the search algorithm, but to restore from a directory we have to use a `Searcher`? Why not the same approach to both restore from paths and directories? I'm sure there's a reason for this but to me it's not clear in the docs.

Reply:
fixed!